lisa/runner.py

```diff
@@ -282,52 +291,41 @@ def _start_loop(self) -> None:
                     f"{task_manager._futures}"
                 )
 
-                for runner in remaining_runners:
-                    while not runner.is_done:
-                        # fetch a task and submit
-                        task = runner.fetch_task()
-                        if task:
-                            # the message may be too long to display.
-                            task_message = str(task)
-                            task_message = (
-                                task_message
-                                if len(task_message) < 200
-                                else f"{task_message[:200]}..."
-                            )
+                while task_manager.has_idle_worker():
```

Comment:
It's better to add explanation on each loop. The explanation should specify what the loop is used for.